pytest_httpserver/httpserver.py

```diff
@@ -426,6 +426,20 @@ def match(self, request: Request) -> bool:
         return not difference
 
 
+def unpack_request_data(r: Request, auto_load: bool):
+    if not auto_load:
+        return r.data
+    data = r.data.decode(r.charset)
+    if not data:
+        return None
+    if r.content_type == 'application/json':
+        return json.loads(data)
+    if r.content_type == 'application/yaml':
```

Comment:
I don't want to add yaml support to pytest-httpserver.
Handling the http responses is one thing and doing the data encoding/decoding is another thing, I want to keep this library to handle the things at http level, and not make this list of possible content types bigger and bigger.
JSON is one exception to this rule, as it is used for REST.

Reply:
BTW we can also make this working by using the JsonMixIn class from werkzeug as it does pretty much the same as this function, it adds methods to the request object which can decode the json data.
https://werkzeug.palletsprojects.com/en/1.0.x/wrappers/#json

pytest-httpserver can be changed in order to work with its own request class inheriting from this mixin also.